### PR TITLE
test(http): Add memory limit parsing tests to improve coverage and handle various input cases in `parseMemoryLimit()` method.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

## Summary by CodeRabbit

- Tests
  - Added a data-driven unit test covering memory-limit parsing across empty, suffixed (k/m/g), numeric, special, whitespace, and edge-case values (including unlimited and zero).
  - Added a shared data provider supplying comprehensive test cases for those scenarios; no production or user-facing changes.
- Chores
  - Adjusted memory-limit parsing behavior to normalize suffix handling (improves robustness; no visible user impact).

### CODE REVIEW

File: tests/http/stateless/ApplicationMemoryTest.php [hunk 1]
```diff
@@ -194,6 +195,21 @@ public function testMemoryLimitHandlesUnlimitedValues(): void
...
+    #[DataProviderExternal(StatelessApplicationProvider::class, 'parseMemoryLimit')]
+    public function testParseMemoryLimitHandlesAllCasesCorrectly(
+        string $input,
+        int $expected,
+        string $assertionMessage,
+    ): void {
+        $app = $this->statelessApplication();
+
+        self::assertSame($expected, self::invokeMethod($app, 'parseMemoryLimit', [$input]), $assertionMessage);
+    }
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply tests/http/stateless/ApplicationMemoryTest.php:211]: Why, use https://[raw.githubusercontent.com/php-forge/support/refs/heads/main/src/TestSupport.php](https://raw.githubusercontent.com/php-forge/support/refs/heads/main/src/TestSupport.php)
